src/main/webapp/js/webrtc_adaptor.js

```diff
@@ -1290,33 +1297,65 @@ export class WebRTCAdaptor
 			console.error("Cannot set local description. Error is: " + error);
 		});
 	}
-
-
-	turnOffLocalCamera() 
-	{
-		if (this.remotePeerConnection != null) {
-
-			var track = this.localStream.getVideoTracks()[0];
-			track.enabled = false;
-		}
-		else {
-			this.callbackError("NoActiveConnection");
-		}
+	initializeDummyFrame(){
+		this.dummyCanvas.getContext('2d').fillRect(0, 0, 320, 240);
+		this.replacementStream = this.dummyCanvas.captureStream();
 	}
 
-	turnOnLocalCamera() 
-	{
-		//If it started in playOnly mode and wants to turn on the camera
-		if(this.localStream == null){
-			this.navigatorUserMedia(this.mediaConstraints, stream =>{
-				this.gotStream(stream);
-			}, false);
-		}
-		else if (this.remotePeerConnection != null) {
-			var track = this.localStream.getVideoTracks()[0];
-			track.enabled = true;
-		}
-	}
+	turnOffLocalCamera(streamId) 
+	 {
+		 //Initialize the first dummy frame for switching.
+		this.initializeDummyFrame();
```

Comment:
If user accidently calls the `turnOffLocalCamera` several times, it creates several canvas and timers

Reply:
Canvas is created on the constructor only once and if a new frame is created previous one is released from the same variable. About timers I have refactored it.